### PR TITLE
Bump Reframe's version

### PR DESCRIPTION
**Pull request by victorusu**
There was also a missing dependency on git on ReFrame.

### CODE REVIEW

File: var/spack/repos/builtin/packages/reframe/package.py [hunk 1]
```diff
@@ -42,6 +43,7 @@ class Reframe(Package):
     depends_on("py-pygelf", when="+gelf", type="run")
     depends_on("py-sphinx", when="+docs", type="build")
     depends_on("py-sphinx-rtd-theme", when="+docs", type="build")
+    depends_on("git", when="@2.0:3.1", type="run")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/reframe/package.py:48]: I assume that also `reframe@master` needs git at runtime. If that is the case then the dependency should read:
```suggestion
    depends_on("git", type="run")
```

**Review by alalazo — CHANGES_REQUESTED** [review]
Please double check the dependency on git. Other than that this LGTM.

File: var/spack/repos/builtin/packages/reframe/package.py (hunk 1 above)
Comment (victorusu) [reply var/spack/repos/builtin/packages/reframe/package.py:48]: Ciao @alalazo, thanks for the review.
Yes, it does require at runtime, but actually the bug was fixed and one won't need `git` for ReFrame 3.2 (https://github.com/eth-cscs/reframe/issues/1464).

I can change to  
```
depends_on("git", type="run")
```
for now, but I will make the change back to 
```
depends_on("git", when="@2.0:3.1", type="run")
```
when 3.2 gets released. 
What do you think?

Comment (alalazo) [reply var/spack/repos/builtin/packages/reframe/package.py:48]: Then I'd just put a comment on top of the current dependency, saying something like:
```python
# ReFrame requires git up to version 3.1, see:
# https://github.com/eth-cscs/reframe/issues/1464
```
I assumed that ReFrame was requiring `git` at runtime to e.g. clone third party repositories but it doesn't seem to be the case.

**Review by alalazo — APPROVED** [review]

**Pull request merged**